File: manual/sphinx/source/Cookbooks/python/PhaseRegion.py
import pyswEOS
from pyswEOS import H2ONaCl
sw=H2ONaCl.cH2ONaCl()

p=200 #bar
T=400 #deg.C
X=0.032 #wt.% NaCl
region = sw.findPhaseRegion(T,p,X)
region_name=sw.getPhaseRegionName(region);
print(" Pressure(bar): ",p)
print(" Temperature(deg.C): ",T)
print(" Salinity (wt.% NaCl): ",X)
print(" Phase region index: ",region)
print(" Phase region name: ",region_name)

File: manual/sphinx/source/Cookbooks/cpp/PhaseRegion.cpp
#include "H2ONaCl.H"
#include <iostream>
H2ONaCl::cH2ONaCl eos;

int main()
{
    H2ONaCl::cH2ONaCl eos;
    double p=200; //bar
    double T=400; //deg.C
    double X=0.032; //wt.% NaCl
    H2ONaCl::PhaseRegion region=eos.findPhaseRegion(T, p, X);
    std::string region_name=eos.getPhaseRegionName(region);
    std::cout<<" Pressure(bar): "<<p<<"\n"
             <<" Temperature(deg.C): "<<T<<"\n"
             <<" Salinity (wt.% NaCl): "<<X<<std::endl;
    std::cout<<" Phase region index: "<<region<<std::endl;
    std::cout<<" Phase region name: "<<region_name<<std::endl;
}